gfx/painter.h

```diff
@@ -0,0 +1,25 @@
+// SPDX-FileCopyrightText: 2022 Mikael Larsson <c.mikael.larsson@gmail.com>
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+#ifndef GFX_PAINTER_H_
+#define GFX_PAINTER_H_
+
+#include "gfx/icanvas.h"
+
+namespace gfx {
+
+class Painter {
+public:
+    Painter(ICanvas &canvas) : canvas_{canvas} {}
+
+    void draw_rect(geom::Rect const &, Color);
```

Comment:
I called it `fill_rect` in `IPainter` as we might want a non-filling variant of it as well. :P